pkg/fanal/analyzer/os/release/release.go

```diff
@@ -61,6 +61,8 @@ func (a osReleaseAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInp
 			family = types.Wolfi
 		case "chainguard":
 			family = types.Chainguard
+		case "azurelinux":
+			family = types.Azure
```

Comment:
We check `etc/mariner-release` file for `mariner` images:
https://github.com/aquasecurity/trivy/blob/b76a7250912cfc028cfef743f0f98cd81b39f8aa/pkg/fanal/analyzer/os/mariner/mariner.go#L29-L37

Does `azure` contain `etc/azure-release` file?
if yes - i think we can update logic of this file.

Reply:
Some Distroless images (e.g., Alpine) include only /etc/os-release, not /etc/xxx-version or /etc/xxx-release. So, I think it's better to use /etc/os-release now.

Reply:
hm... i didn't know about this.
then, i think we need to move logic for `mariner-release` to `os-release`

Reply:
Sorry I was confused. [Distroless](https://github.com/distroless) didn't even contain`/etc/os-release.`, and then included `/etc/os-release` and `/etc/alpine-release`.
https://github.com/aquasecurity/trivy/issues/1975

Here is another discussion about `/etc/os-release`.
https://github.com/aquasecurity/trivy/issues/3485

So, it's anyway good to get a version from `/etc/os-release`.

Reply:
Both Mariner and Azure Linux include an /etc/os-release file. Mariner has a /etc/mariner-release and Azure has /etc/azurelinux-release file.

I've removed the mariner-release parsing and updated Mariner to use /etc/os-release parsing. This means that the build ID is removed from the distro in package URLs:

before that change > `ExternalRef: PACKAGE-MANAGER purl pkg:rpm/cbl-mariner/curl@8.5.0-1.cm2?arch=x86_64&distro=cbl-mariner-2.0.20240123`
after that change > `ExternalRef: PACKAGE-MANAGER purl pkg:rpm/cbl-mariner/curl@8.5.0-1.cm2?arch=x86_64&distro=cbl-mariner-2.0`

The PURL spec doesn't say what the distro qualifier should be for RPMs. It seems unnecessary to include the build ID since the package can be identified via the other fields.

Reply:
We don't need the assembly ID to detect vulnerabilities.
Looks like we don't need it.

@knqyf263 do you have any objections?

@tofay Can you add tests for detection of  `1.0`, `2.0` and `3.0` in [release_test.go](https://github.com/aquasecurity/trivy/blob/main/pkg/fanal/analyzer/os/release/release_test.go)?

Reply:
>Looks like we don't need it.

+1

Reply:
> @tofay Can you add tests for detection of 1.0, 2.0 and 3.0 in [release_test.go](https://github.com/aquasecurity/trivy/blob/main/pkg/fanal/analyzer/os/release/release_test.go)?

Done